tech/post/2017-09-23-Running-the-kubernetes-on-OpenStack-step-by-step.md

```diff
@@ -0,0 +1,601 @@
+Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
+default         192.168.98.122  0.0.0.0         UG    0      0        0 ens3
+172.17.0.0      *               255.255.0.0     U     0      0        0 docker0
+192.168.98.0    *               255.255.255.0   U     0      0        0 ens3
+root@ubuntu:/opt/kube# ifconfig
+docker0   Link encap:Ethernet  HWaddr 02:42:85:0e:87:c7  
+          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
+          UP BROADCAST MULTICAST  MTU:1500  Metric:1
+          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0 
+          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
+
+ens3      Link encap:Ethernet  HWaddr fa:16:3e:cc:95:3c  
+          inet addr:192.168.98.40  Bcast:192.168.98.255  Mask:255.255.255.0
+          inet6 addr: fe80::f816:3eff:fecc:953c/64 Scope:Link
+          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
+          RX packets:4116103 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:235816 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:1000 
+          RX bytes:1194455376 (1.1 GB)  TX bytes:23193366 (23.1 MB)
+
+lo        Link encap:Local Loopback  
+          inet addr:127.0.0.1  Mask:255.0.0.0
+          inet6 addr: ::1/128 Scope:Host
+          UP LOOPBACK RUNNING  MTU:65536  Metric:1
+          RX packets:160 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:160 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:1 
+          RX bytes:11840 (11.8 KB)  TX bytes:11840 (11.8 KB)
+
+root@ubuntu:/opt/kube# dhclient ens3
+root@ubuntu:/opt/kube# route
+Kernel IP routing table
+Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
+default         192.168.98.122  0.0.0.0         UG    0      0        0 ens3
+169.254.169.254 192.168.98.30   255.255.255.255 UGH   0      0        0 ens3
+172.17.0.0      *               255.255.0.0     U     0      0        0 docker0
+192.168.98.0    *               255.255.255.0   U     0      0        0 ens3
+root@ubuntu:/opt/kube# 
+
+```
+
+Login this instance and check the nova metadata service:
+```bash
+root@ubuntu:/opt/stack/karbor# ssh ubuntu@publicsite=192.168.98.40
+ubuntu@ubuntu:~$ curl 169.254.169.254/1.0/meta-data
+ami-id
+ami-launch-index
+ami-manifest-path
+hostname
+instance-id
+local-ipv4
+reservation-id
+security-groups
+ubuntu@ubuntu:~$
+```
+
+## Install a single node kubernetes on Nova instance
+
+Requirements:
+
+OS: Ubuntu 16.04.2 LTS
+CPU: 2 vCPUs
+Memory: 4G
+Storage: 10G
+
+Docker:  1.12.6
+Go: 1.9
+Etcd: v3.2.6
+Kubernetes:master
+
+###  1 Install tools:
+Install git: 
+```bash
+sudo apt-get install git
+```
+
+Install gcc: 
+```bash
+sudo apt-get install gcc
+```
+
+###  2 Install docker:
+```bash
+root@ubuntu:/#apt-get update
+root@ubuntu:/#apt-get upgrade
+
+root@ubuntu:/#apt-get install docker.io
```

Comment:
I recommend to use the following way of docker installation:
wget -qO- https://get.docker.com | sh

Reply:
Done.